### PR TITLE
Update pinned dependencies

### PR DESCRIPTION
**Pull request by Zac-HD**
Automatically update pinned dependencies

### CODE REVIEW

File: tooling/src/hypothesistooling/__main__.py [hunk 1]
```diff
@@ -435,9 +435,9 @@ def run_tox(task, version, *args):
     "3.9": "3.9.19",
     "3.10": "3.10.14",
     "3.11": "3.11.9",
-    "3.12": "3.12.3",
+    "3.12": "3.12.4",
     "3.13": "3.13.0b2",
-    "3.14": "3.14-dev",
+    "3.14": "3.14t-dev",
     "pypy3.8": "pypy3.8-7.3.11",
```
Comment (tybug) [inline tooling/src/hypothesistooling/__main__.py:441]: what does the `t` denote in `3.14t-dev` 🤔 ? 

```python
from packaging.version import Version

# packaging.version.InvalidVersion: Invalid version: '3.14t-dev'
Version("3.14t-dev")
```

Comment (Zac-HD) [reply tooling/src/hypothesistooling/__main__.py:441]: It's [the free-threading build](https://github.com/pyenv/pyenv/blob/master/plugins/python-build/share/python-build/3.14t-dev)!  We're [not actually running 3.14 in CI yet](https://github.com/HypothesisWorks/hypothesis/blob/d8c17832141f587fec9ef89895ed01a4e9c1650d/.github/workflows/main.yml#L52-L54), but we should probably / eventually:

- [ ] fix our auto-update script so it defaults to the standard build
- [ ] start running CI on 3.14 standard
- [ ] add another set of builds for the free-threading version and run that too

File: tooling/src/hypothesistooling/__main__.py [hunk 2]
```diff
@@ -435,9 +435,9 @@ def run_tox(task, version, *args):
     "3.9": "3.9.19",
     "3.10": "3.10.14",
     "3.11": "3.11.9",
-    "3.12": "3.12.3",
+    "3.12": "3.12.4",
     "3.13": "3.13.0b2",
```
Comment (jobh) [inline tooling/src/hypothesistooling/__main__.py:439]: It looks like we may need

```suggestion
    "3.13": "3.13.0b3",
```

File: hypothesis-python/tests/cover/test_lookup.py [hunk 3]
```diff
@@ -112,7 +115,10 @@ def test_typing_Type_Union(ex):
 @pytest.mark.parametrize(
     "typ",
     [
-        collections.abc.ByteString,
+        pytest.param(
+            getattr(collections.abc, "ByteString", ...),
+            marks=pytest.mark.skipif(sys.version_info[:2] == (3, 14), reason="removed"),
```
Comment (jobh) [inline hypothesis-python/tests/cover/test_lookup.py:120]: I see this is work in progress, but
```suggestion
            marks=pytest.mark.skipif(sys.version_info[:2] >= (3, 14), reason="removed"),
```
?

Comment (Zac-HD) [reply hypothesis-python/tests/cover/test_lookup.py:120]: Yeah, looks like I just changed the wrong comparison late at night 🤦‍♂️

**Pull request merged**